socorro/unittest/external/fs/test_tar_crashstore.py

```diff
@@ -1,85 +1,143 @@
 
         processed_crash = {
             'crash_id': '091204bd-87c0-42ba-8f58-554492141212',
             'payload': 'nothing to see here',
             'some_date': datetime(1960, 5, 4, 15, 10)
         }
-        processed_crash_as_string = json.dumps(
+        processed_crash_as_str = json.dumps(
             processed_crash,
-            default=tar_store.stringify_datetimes
+            default=TarFileWritingCrashStore.stringify_datetimes
         )
 
         # the call to be tested
         tar_store.save_processed(processed_crash)
 
-        # this is what should have happened
-        ok_(hasattr(tar_store, 'tar_file'))
-        tar_store.tarfile_module.open.assert_called_once_with(
-            config.tarball_name,
-            'w'
-        )
-        tar_store.gzip_module.open.assert_called_once_with(
-            '/tmp/091204bd-87c0-42ba-8f58-554492141212.jsonz',
-            'w',
-            9
-        )
-        mocked_file_handle = tar_store.gzip_module.open.return_value
-        mocked_file_handle.write.assert_called_once_with(
-            processed_crash_as_string
-        )
-        mocked_file_handle.close.assert_called_once_with()
-        tar_store.tar_file.add.assert_called_once_with(
-            '/tmp/091204bd-87c0-42ba-8f58-554492141212.jsonz',
-            '09/12/091204bd-87c0-42ba-8f58-554492141212.jsonz'
+        # this try to get it back
+        tar_store.close()
+
+        result_tar_fp = tarfile.TarFile('/tmp/a_tarball_name.tar')
+        result_gzip_fp = gzip.GzipFile(
+            fileobj=result_tar_fp.extractfile(
+                '091204bd-87c0-42ba-8f58-554492141212.jsonz'
+            )
         )
-        tar_store.os_module.unlink.assert_called_once_with(
-            '/tmp/091204bd-87c0-42ba-8f58-554492141212.jsonz'
+        reconstituted_processed_crash_as_str = result_gzip_fp.read().strip()
+
+        self.assertEqual(
+            processed_crash_as_str,
+            reconstituted_processed_crash_as_str
         )
 
 
+#==============================================================================
+class TestTarFileSequentialReadingCrashStorage(TestCase):
+
+    #--------------------------------------------------------------------------
+    def setUp(self):
+        super(TestTarFileSequentialReadingCrashStorage, self).setUp()
+
+    #--------------------------------------------------------------------------
+    def _get_config(self):
+        config = DotDict()
+        config.logger = Mock()
+        config.tarball_name = '/tmp/a_tarball_name.tar'
+        config.tarfile_module = tarfile
+        config.gzip_module = gzip
+        config.redactor_class = Redactor
+        config.forbidden_keys = ''
+
+        return config
+
+    #--------------------------------------------------------------------------
+    def test_get_processed(self):
+        config = self._get_config()
+        config.tarfile_module = tarfile
+        config.gzip_module = gzip
```

Comment:
why these two lines? Isn't it already that by default (by looking at `_get_config()`)